src/d3a/models/state.py

```diff
@@ -56,17 +56,33 @@ def get_state(self) -> Dict:
 
     @abstractmethod
     def restore_state(self, state_dict: Dict):
-        pass
+        """Update the state of the device using the provided dictionary."""
 
     @abstractmethod
-    def delete_past_state_values(self, time_slot):
-        pass
+    def delete_past_state_values(self, current_time_slot: DateTime):
+        """Delete the state of the device before the given time slot."""
 
     def __str__(self):
         return self.__class__.__name__
 
 
-class ConsumptionState(StateInterface):
+class ProsumptionInterface(StateInterface, ABC):
+    """Interface with common methods/variables shared by consumption and production devices."""
+
+    def __init__(self):
+        # Actual energy consumed/produced by the device at specific market slots
+        self._real_energy: Dict[DateTime, float] = {}
+
+    def set_real_energy_kWh(self, energy_kWh: float, time_slot: DateTime) -> None:
+        """Set the actual energy consumed/produced by the device in the given market slot."""
+        self._real_energy[time_slot] = energy_kWh
+
+    def get_real_energy_kWh(self, time_slot: DateTime) -> float:
```

Comment:
I just realise, that my current story (D3ASIM-3537) is indeed dependant on this one, because of your changes in the state.py here. Unfortunately, there is no consistent naming for the real energy, but I would vote to name it `energy_measurement` actually. Here in this story these are just simulated. Do you agree?

Reply:
I agree about the naming issue. The problem is that all the previous values we had were "forecasts" but their names make it look like they're real energy readings.
That's why I was looking for a way to make this new energy look different from the others (and adding `_real_` to the name was a way to achieve this).
I'm worried that calling it `energy_measurement` will not make it clear that this is different from any other energy we currently have. For example, suppose you read:
- `self.available_energy`
- `self.energy_measurement`
How can you know that these two are actually talking about two very different types of energy (forecast vs real)?

Reply:
Yeah. I agree. I think it is time to change these. If I correctly remember, this is on the table since forever. This will make also other coders help more to understand that there is a big difference between the forecast and the measurement. What do you think @faizan2590 ?

Reply:
My proposal is the following renaming that we could put into a separate story:
`ConsumptionState._energy_requirement_Wh` --> `ConsumptionState._energy_consumption_forecast_kWh`
(this will contain a unit change and also includes renaming the corresponding set and get methods)
The PV state is already aligned with this.
With this you could name the `_real_energy ` into `_energy_consumption_measurement_kWh` and `_energy_production_measurement_kWh`
What do you think? Also tagging @faizan2590 again.

Reply:
For this PR here I would be strongly for adding the unit to the variable, too (so to name if `self._real_energy_kWh`)

Reply:
> For this PR here I would be strongly for adding the unit to the variable, too (so to name if self._real_energy_kWh)

I agree, will do it now.
I also think that the renaming you suggest would work. I am thinking about some bigger refactoring for the energy values (to create a simpler interface shared by the states) but this is just an idea for the moment.

Reply:
> My proposal is the following renaming that we could put into a separate story:
ConsumptionState._energy_requirement_Wh --> ConsumptionState._energy_consumption_forecast_kWh
(this will contain a unit change and also includes renaming the corresponding set and get methods)
The PV state is already aligned with this.
With this you could name the _real_energy into _energy_consumption_measurement_kWh and _energy_production_measurement_kWh
What do you think? Also tagging @faizan2590 again.


Sorry @hannesdiedrich for the late response, but I totally agree with your proposal.